esphome/components/jsn_sr04t/jsn_sr04t.cpp

```diff
@@ -0,0 +1,54 @@
+#include "jsn_sr04t.h"
+#include "esphome/core/helpers.h"
+#include "esphome/core/log.h"
+
+// Very basic support for JSN_SR04T V3.0 distance sensor in mode 2
+
+namespace esphome {
+namespace jsn_sr04t {
+
+static const char *const TAG = "jsn_sr04t.sensor";
+
+void Jsnsr04tComponent::update() {
+  this->write_byte(0x55);
+  ESP_LOGV(TAG, "Request read out from sensor");
+
+  while (this->available() > 0) {
+    uint8_t data;
+    this->read_byte(&data);
+
+    ESP_LOGV(TAG, "Read byte from sensor: %x", data);
+
+    if (this->buffer_.empty() && data != 0xFF)
+      continue;
+
+    this->buffer_.push_back(data);
+    if (this->buffer_.size() == 4)
+      check_buffer_();
+  }
+}
```

Comment:
Read requests should happen in `update`, and checking for response bytes should happen in the `loop` in case there is a delay receiving the response.
```suggestion
void Jsnsr04tComponent::update() {
  this->write_byte(0x55);
  ESP_LOGV(TAG, "Request read out from sensor");
}
void Jsnsr04tComponent::loop() {
  while (this->available() > 0) {
    uint8_t data;
    this->read_byte(&data);

    ESP_LOGV(TAG, "Read byte from sensor: %x", data);

    if (this->buffer_.empty() && data != 0xFF)
      continue;

    this->buffer_.push_back(data);
    if (this->buffer_.size() == 4)
      check_buffer_();
  }
}
```